cloudevents/working-drafts/dds-format.md

```diff
@@ -0,0 +1,243 @@
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are mapped to the IDL
+type-system used by DDS. This specification explicitly maps each attribute.
+
+### 2.1 Type System Mapping
+
+The CloudEvents type system MUST be mapped to DDS types as follows.
+
+| CloudEvents   | DDS                                                                   |
+| ------------- | ---------------------------------------------------------------------- |
+| Boolean       | Boolean                                             |
+| Integer       | Int32                                                 |
+| String        | String (255)                                              |
+| Binary        | Bytes (100)                                               |
+| URI           | String (255) following [RFC 3986 §4.3][rfc3986-section43] |
+| URI-reference | String (255) following [RFC 3986 §4.1][rfc3986-section41] |
+| Timestamp     | seconds: Int64 nanoseconds: UInt32     |
+
+Extension specifications MAY define secondary mapping rules for the values of
+attributes they define, but MUST also include the previously defined primary
+mapping.
+
+### 2.2 OPTIONAL Attributes
+
+CloudEvents Spec defines OPTIONAL attributes. The set of possible Attribute Types and Values
+for optional attributes are defined in the [DDS Event Format][dds-event-format] as follows:
+
+```xml
+<enum name="AttributeType">
+   <enumerator name="BOOL"/>
+   <enumerator name="INT32"/>
+   <enumerator name="STRING"/>
+   <enumerator name="BYTES"/>
+   <enumerator name="URI"/>
+   <enumerator name="URI_REF"/>
+   <enumerator name="TIMESTAMP"/>
+</enum>
+	  
+<union name="AttributeValue" extensibility="final">
+   <discriminator type="nonBasic" nonBasicTypeName="io::cloudevents::AttributeType"/>
+   <case>
+      <caseDiscriminator value="(io::cloudevents::BOOL)"/>
+      <member name="ce_boolean" type="boolean"/>
+   </case>
+   <case>
+      <caseDiscriminator value="(io::cloudevents::INT32)"/>
+      <member name="ce_integer" type="int32"/>
+   </case>
+   <case>
+      <caseDiscriminator value="(io::cloudevents::STRING)"/>
+      <member name="ce_string" type="string" stringMaxLength="255"/>
+   </case>
+   <case>
+      <caseDiscriminator value="(io::cloudevents::BYTES)"/>
+      <member name="ce_bytes" type="byte" sequenceMaxLength="100"/>
+   </case>
+   <case>
+      <caseDiscriminator value="(io::cloudevents::URI)"/>
+      <member name="ce_uri" type="nonBasic" nonBasicTypeName="io::cloudevents::URI"/>
+   </case>
+   <case>
+      <caseDiscriminator value="(io::cloudevents::URI_REF)"/>
+      <member name="ce_uri_reference" type="nonBasic" nonBasicTypeName="io::cloudevents::URI_Ref"/>
+   </case>
+   <case>
+      <caseDiscriminator value="(io::cloudevents::TIMESTAMP)"/>
+      <member name="ce_timestamp" type="nonBasic" nonBasicTypeName="io::cloudevents::Timestamp"/>
+   </case>
+</union>
+
+```
+
+Based on the above, the type definition for optional Attributes is as follows:
```

Comment:
I feel like I'm missing something... what in the example below indicates that the attribute being defined is optional vs mandatory?  Also is `Attribute` (in `<struct name="Attribute">`) the name of the attribute? If so, perhaps call it something less ambiguous, like `MyExtension` - otherwise it looks like a keyword and not a user defined name.

Reply:
I changed "Attribute" -> "DDS-CE-Attribute" and "Attributes" -> "DDS-CE-Attributes". Does that read any better to you?